lte/gateway/c/session_manager/LocalSessionManagerHandler.cpp

```diff
@@ -242,54 +242,73 @@ void LocalSessionManagerHandlerImpl::CreateSession(
   set_sentry_transaction("CreateSession");
   auto& request_cpy = *request;
   PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request_cpy));
-  enforcer_->get_event_base().runInEventBaseThread(
-      [this, context, response_callback, request_cpy]() {
-        SessionConfig cfg(request_cpy);
-        const std::string& imsi = cfg.get_imsi();
-        const auto& session_id  = id_gen_.gen_session_id(imsi);
-        log_create_session(cfg);
-        if (pipelined_state_ != READY) {
-          MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
-                      << " apn=" << cfg.common_context.apn()
-                      << " since PipelineD is still setting up.";
-          send_local_create_session_response(
-              Status(grpc::UNAVAILABLE, "PipelineD is not ready"), session_id,
-              response_callback);
-          return;
-        }
-        if (!session_store_.is_ready()) {
-          MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
-                      << " apn=" << cfg.common_context.apn()
-                      << " since SessionStore (Redis) is unavailable.";
-          send_local_create_session_response(
-              Status(grpc::UNAVAILABLE, "Storage backend is not available"),
-              session_id, response_callback);
-          return;
-        }
+  enforcer_->get_event_base().runInEventBaseThread([this, context,
+                                                    response_callback,
+                                                    request_cpy]() {
+    SessionConfig cfg(request_cpy);
+    const std::string& imsi = cfg.get_imsi();
+    const auto& session_id  = id_gen_.gen_session_id(imsi);
+    log_create_session(cfg);
+    if (pipelined_state_ != READY) {
+      MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
+                  << " apn=" << cfg.common_context.apn()
+                  << " since PipelineD is still setting up.";
+      send_local_create_session_response(
+          Status(grpc::UNAVAILABLE, "PipelineD is not ready"), session_id,
+          response_callback);
+      return;
+    }
+    if (!session_store_.is_ready()) {
+      MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
+                  << " apn=" << cfg.common_context.apn()
+                  << " since SessionStore (Redis) is unavailable.";
+      send_local_create_session_response(
+          Status(grpc::UNAVAILABLE, "Storage backend is not available"),
+          session_id, response_callback);
+      return;
+    }
+    const CommonSessionContext common = cfg.common_context;
+    if (common.rat_type() != TGPP_WLAN && common.rat_type() != TGPP_LTE) {
```

Comment:
why don't we put this condition at the very top?

Reply:
good call will fix that